# CMU Movies Summary Corpus

- Authors: Zaynab, Lylia, Ali, Christian, Yassin

---

## Tasks

1. **Select Project & Initial Analyses**:
   1. Agree on a project proposal with team members.
   2. Perform initial analyses to verify feasibility of the proposed project, including any additional data.
   3. Acquaint yourself with the provided data, preprocess it, and perform descriptive statistics.

2. **Pipeline & Data Description**:
   1. Create a pipeline for data handling and preprocessing, documented in the notebook.
   2. Describe the relevant aspects of the data, including:
      1. Handling the size of the data.
      2. Understanding the data (formats, distributions, missing values, correlations, etc.).
      3. Considering data enrichment, filtering, and transformation according to project needs.
   3. Develop a plan for methods to be used, with essential mathematical details.
   4. Outline a plan for analysis and communication, discussing alternative approaches considered.

3. **GitHub Repository & Deliverables**:
   1. Create a public GitHub repository named `ada-2023-project-<team>` under the `epfl-ada` GitHub organization. ✅
   2. Ensure the repository contains:
      1. **README.md** file with:
         1. **Title**: Project title.
         2. **Abstract**: 150-word description of the project idea, goals, and motivation.
         3. **Research Questions**: List of research questions to address.
         4. **Proposed Additional Datasets**: Description of additional datasets, expected management, and feasibility analysis.
         5. **Methods**: Methods to be used in the project.
         6. **Proposed Timeline**: Timeline for the project.
         7. **Organization within the Team**: Internal milestones leading to Milestone P3.
         8. **Questions for TAs (optional)**: Any questions for the teaching assistants.
      2. **Code for Initial Analyses**: Structured code for initial analyses and data handling pipelines.
      3. **Notebook** presenting initial results, including:
         1. Main results and descriptive analysis.
         2. External scripts/modules for implementing core logic, to be called from the notebook.

---


## Table of Contents
- [1. Zaynab's part](##Zaynab's-part)
- [2. Lylia's part](##Lylia's-part)
- [3. Ali's part](##Ali's-part)
- [4. Cristians's part](##Christian's-part)
- [5. Yassin's part](##Yassin's-part)

---

### Library importation

In [1]:
import pandas as pd 
import numpy as np 
import matplotlib.pyplot as plt
import seaborn as sns

### data importation

In [2]:
DATA_PATH='./data/MovieSummaries/'

### movie metadata

In [3]:
movie_columns = [
    'WikipediaMovieID', 'FreebaseMovieID', 'MovieName', 'ReleaseDate', 
    'BoxOfficeRevenue', 'Runtime', 'Languages', 'Countries', 'Genres'
]


movie_metadata = pd.read_csv(DATA_PATH+'movie.metadata.tsv', sep='\t', names=movie_columns)

movie_metadata

WikipediaMovieID FreebaseMovieID  \
0                975900       /m/03vyhn   
1               3196793       /m/08yl5d   
2              28463795      /m/0crgdbh   
3               9363483      /m/0285_cd   
4                261236       /m/01mrr1   
...                 ...             ...   
81736          35228177      /m/0j7hxnt   
81737          34980460      /m/0g4pl34   
81738           9971909      /m/02pygw1   
81739            913762       /m/03pcrp   
81740          12476867      /m/02w7zz8   

                                               MovieName ReleaseDate  \
0                                         Ghosts of Mars  2001-08-24   
1      Getting Away with Murder: The JonBenét Ramsey ...  2000-02-16   
2                                            Brun bitter        1988   
3                                       White Of The Eye        1987   
4                                      A Woman in Flames        1983   
...                                                  ...         ...   
81736                           Mermaids: The Body Found  2011-03-19   
81737                                            Knuckle  2011-01-21   
81738                                  Another Nice Mess  1972-09-22   
81739  The Super Dimension Fortress Macross II: Lover...  1992-05-21   
81740                                            Spliced        2002   

       BoxOfficeRevenue  Runtime                           Languages  \
0            14010832.0     98.0  {"/m/02h40lc": "English Language"}   
1                   NaN     95.0  {"/m/02h40lc": "English Language"}   
2                   NaN     83.0  {"/m/05f_3": "Norwegian Language"}   
3                   NaN    110.0  {"/m/02h40lc": "English Language"}   
4                   NaN    106.0   {"/m/04306rv": "German Language"}   
...                 ...      ...                                 ...   
81736               NaN    120.0  {"/m/02h40lc": "English Language"}   
81737               NaN     96.0  {"/m/02h40lc": "English Language"}   
81738               NaN     66.0  {"/m/02h40lc": "English Language"}   
81739               NaN    150.0   {"/m/03_9r": "Japanese Language"}   
81740               NaN     86.0  {"/m/02h40lc": "English Language"}   

                                               Countries  \
0              {"/m/09c7w0": "United States of America"}   
1              {"/m/09c7w0": "United States of America"}   
2                                 {"/m/05b4w": "Norway"}   
3                         {"/m/07ssc": "United Kingdom"}   
4                                {"/m/0345h": "Germany"}   
...                                                  ...   
81736          {"/m/09c7w0": "United States of America"}   
81737  {"/m/03rt9": "Ireland", "/m/07ssc": "United Ki...   
81738          {"/m/09c7w0": "United States of America"}   
81739                              {"/m/03_3d": "Japan"}   
81740                            {"/m/0d060g": "Canada"}   

                                                  Genres  
0      {"/m/01jfsb": "Thriller", "/m/06n90": "Science...  
1      {"/m/02n4kr": "Mystery", "/m/03bxz7": "Biograp...  
2      {"/m/0lsxr": "Crime Fiction", "/m/07s9rl0": "D...  
3      {"/m/01jfsb": "Thriller", "/m/0glj9q": "Erotic...  
4                                {"/m/07s9rl0": "Drama"}  
...                                                  ...  
81736                            {"/m/07s9rl0": "Drama"}  
81737  {"/m/03bxz7": "Biographical film", "/m/07s9rl0...  
81738       {"/m/06nbt": "Satire", "/m/01z4y": "Comedy"}  
81739  {"/m/06n90": "Science Fiction", "/m/0gw5n2f": ...  
81740  {"/m/01jfsb": "Thriller", "/m/03npn": "Horror"...  

[81741 rows x 9 columns]

In [4]:
movie_metadata['ReleaseDate']=movie_metadata['ReleaseDate'].str.extract(r'(\d{4})')
movie_metadata['ReleaseDate'] = movie_metadata['ReleaseDate'].astype(float)



movie_metadata

WikipediaMovieID FreebaseMovieID  \
0                975900       /m/03vyhn   
1               3196793       /m/08yl5d   
2              28463795      /m/0crgdbh   
3               9363483      /m/0285_cd   
4                261236       /m/01mrr1   
...                 ...             ...   
81736          35228177      /m/0j7hxnt   
81737          34980460      /m/0g4pl34   
81738           9971909      /m/02pygw1   
81739            913762       /m/03pcrp   
81740          12476867      /m/02w7zz8   

                                               MovieName  ReleaseDate  \
0                                         Ghosts of Mars       2001.0   
1      Getting Away with Murder: The JonBenét Ramsey ...       2000.0   
2                                            Brun bitter       1988.0   
3                                       White Of The Eye       1987.0   
4                                      A Woman in Flames       1983.0   
...                                                  ...          ...   
81736                           Mermaids: The Body Found       2011.0   
81737                                            Knuckle       2011.0   
81738                                  Another Nice Mess       1972.0   
81739  The Super Dimension Fortress Macross II: Lover...       1992.0   
81740                                            Spliced       2002.0   

       BoxOfficeRevenue  Runtime                           Languages  \
0            14010832.0     98.0  {"/m/02h40lc": "English Language"}   
1                   NaN     95.0  {"/m/02h40lc": "English Language"}   
2                   NaN     83.0  {"/m/05f_3": "Norwegian Language"}   
3                   NaN    110.0  {"/m/02h40lc": "English Language"}   
4                   NaN    106.0   {"/m/04306rv": "German Language"}   
...                 ...      ...                                 ...   
81736               NaN    120.0  {"/m/02h40lc": "English Language"}   
81737               NaN     96.0  {"/m/02h40lc": "English Language"}   
81738               NaN     66.0  {"/m/02h40lc": "English Language"}   
81739               NaN    150.0   {"/m/03_9r": "Japanese Language"}   
81740               NaN     86.0  {"/m/02h40lc": "English Language"}   

                                               Countries  \
0              {"/m/09c7w0": "United States of America"}   
1              {"/m/09c7w0": "United States of America"}   
2                                 {"/m/05b4w": "Norway"}   
3                         {"/m/07ssc": "United Kingdom"}   
4                                {"/m/0345h": "Germany"}   
...                                                  ...   
81736          {"/m/09c7w0": "United States of America"}   
81737  {"/m/03rt9": "Ireland", "/m/07ssc": "United Ki...   
81738          {"/m/09c7w0": "United States of America"}   
81739                              {"/m/03_3d": "Japan"}   
81740                            {"/m/0d060g": "Canada"}   

                                                  Genres  
0      {"/m/01jfsb": "Thriller", "/m/06n90": "Science...  
1      {"/m/02n4kr": "Mystery", "/m/03bxz7": "Biograp...  
2      {"/m/0lsxr": "Crime Fiction", "/m/07s9rl0": "D...  
3      {"/m/01jfsb": "Thriller", "/m/0glj9q": "Erotic...  
4                                {"/m/07s9rl0": "Drama"}  
...                                                  ...  
81736                            {"/m/07s9rl0": "Drama"}  
81737  {"/m/03bxz7": "Biographical film", "/m/07s9rl0...  
81738       {"/m/06nbt": "Satire", "/m/01z4y": "Comedy"}  
81739  {"/m/06n90": "Science Fiction", "/m/0gw5n2f": ...  
81740  {"/m/01jfsb": "Thriller", "/m/03npn": "Horror"...  

[81741 rows x 9 columns]

### character metadata

In [5]:
character_columns = [
    'WikipediaMovieID', 'FreebaseMovieID', 'ReleaseDate', 'CharacterName',
    'ActorDOB', 'ActorGender', 'ActorHeight', 'ActorEthnicity', 
    'ActorName', 'ActorAgeAtRelease', 'FreebaseCharacterActorMapID',
    'FreebaseCharacterID', 'FreebaseActorID'
]

character_metadata = pd.read_csv(DATA_PATH+'character.metadata.tsv', sep='\t', names=character_columns)


character_metadata

WikipediaMovieID FreebaseMovieID ReleaseDate  \
0                 975900       /m/03vyhn  2001-08-24   
1                 975900       /m/03vyhn  2001-08-24   
2                 975900       /m/03vyhn  2001-08-24   
3                 975900       /m/03vyhn  2001-08-24   
4                 975900       /m/03vyhn  2001-08-24   
...                  ...             ...         ...   
450664            913762       /m/03pcrp  1992-05-21   
450665            913762       /m/03pcrp  1992-05-21   
450666          28308153      /m/0cp05t9        1957   
450667          28308153      /m/0cp05t9        1957   
450668          28308153      /m/0cp05t9        1957   

                     CharacterName    ActorDOB ActorGender  ActorHeight  \
0                         Akooshay  1958-08-26           F        1.620   
1       Lieutenant Melanie Ballard  1974-08-15           F        1.780   
2              Desolation Williams  1969-06-15           M        1.727   
3               Sgt Jericho Butler  1967-09-12           M        1.750   
4                  Bashira Kincaid  1977-09-25           F        1.650   
...                            ...         ...         ...          ...   
450664                      Elensh     1970-05           F          NaN   
450665                      Hibiki  1965-04-12           M          NaN   
450666                         NaN  1941-11-18           M        1.730   
450667                         NaN         NaN         NaN          NaN   
450668                         NaN         NaN         NaN          NaN   

       ActorEthnicity           ActorName  ActorAgeAtRelease  \
0                 NaN      Wanda De Jesus               42.0   
1          /m/044038p  Natasha Henstridge               27.0   
2             /m/0x67            Ice Cube               32.0   
3                 NaN       Jason Statham               33.0   
4                 NaN         Clea DuVall               23.0   
...               ...                 ...                ...   
450664            NaN  Dorothy Elias-Fahn                NaN   
450665            NaN       Jonathan Fahn               27.0   
450666      /m/02w7gg      David Hemmings               15.0   
450667            NaN    Roberta Paterson                NaN   
450668            NaN         John Rogers                NaN   

       FreebaseCharacterActorMapID FreebaseCharacterID FreebaseActorID  
0                       /m/0bgchxw          /m/0bgcj3x      /m/03wcfv7  
1                        /m/0jys3m          /m/0bgchn4       /m/0346l4  
2                        /m/0jys3g          /m/0bgchn_      /m/01vw26l  
3                       /m/02vchl6          /m/0bgchnq       /m/034hyc  
4                       /m/02vbb3r          /m/0bgchp9       /m/01y9xg  
...                            ...                 ...             ...  
450664                  /m/0kr406c          /m/0kr406h       /m/0b_vcv  
450665                  /m/0kr405_          /m/0kr4090       /m/0bx7_j  
450666                  /m/0g8ngmc                 NaN       /m/022g44  
450667                  /m/0g8ngmj                 NaN      /m/0g8ngmm  
450668                  /m/0g8ngmw                 NaN      /m/0btz19d  

[450669 rows x 13 columns]

In [6]:
character_metadata['ReleaseDate']=character_metadata['ReleaseDate'].str.extract(r'(\d{4})')
character_metadata['ReleaseDate'] = character_metadata['ReleaseDate'].astype(float)


character_metadata

WikipediaMovieID FreebaseMovieID  ReleaseDate  \
0                 975900       /m/03vyhn       2001.0   
1                 975900       /m/03vyhn       2001.0   
2                 975900       /m/03vyhn       2001.0   
3                 975900       /m/03vyhn       2001.0   
4                 975900       /m/03vyhn       2001.0   
...                  ...             ...          ...   
450664            913762       /m/03pcrp       1992.0   
450665            913762       /m/03pcrp       1992.0   
450666          28308153      /m/0cp05t9       1957.0   
450667          28308153      /m/0cp05t9       1957.0   
450668          28308153      /m/0cp05t9       1957.0   

                     CharacterName    ActorDOB ActorGender  ActorHeight  \
0                         Akooshay  1958-08-26           F        1.620   
1       Lieutenant Melanie Ballard  1974-08-15           F        1.780   
2              Desolation Williams  1969-06-15           M        1.727   
3               Sgt Jericho Butler  1967-09-12           M        1.750   
4                  Bashira Kincaid  1977-09-25           F        1.650   
...                            ...         ...         ...          ...   
450664                      Elensh     1970-05           F          NaN   
450665                      Hibiki  1965-04-12           M          NaN   
450666                         NaN  1941-11-18           M        1.730   
450667                         NaN         NaN         NaN          NaN   
450668                         NaN         NaN         NaN          NaN   

       ActorEthnicity           ActorName  ActorAgeAtRelease  \
0                 NaN      Wanda De Jesus               42.0   
1          /m/044038p  Natasha Henstridge               27.0   
2             /m/0x67            Ice Cube               32.0   
3                 NaN       Jason Statham               33.0   
4                 NaN         Clea DuVall               23.0   
...               ...                 ...                ...   
450664            NaN  Dorothy Elias-Fahn                NaN   
450665            NaN       Jonathan Fahn               27.0   
450666      /m/02w7gg      David Hemmings               15.0   
450667            NaN    Roberta Paterson                NaN   
450668            NaN         John Rogers                NaN   

       FreebaseCharacterActorMapID FreebaseCharacterID FreebaseActorID  
0                       /m/0bgchxw          /m/0bgcj3x      /m/03wcfv7  
1                        /m/0jys3m          /m/0bgchn4       /m/0346l4  
2                        /m/0jys3g          /m/0bgchn_      /m/01vw26l  
3                       /m/02vchl6          /m/0bgchnq       /m/034hyc  
4                       /m/02vbb3r          /m/0bgchp9       /m/01y9xg  
...                            ...                 ...             ...  
450664                  /m/0kr406c          /m/0kr406h       /m/0b_vcv  
450665                  /m/0kr405_          /m/0kr4090       /m/0bx7_j  
450666                  /m/0g8ngmc                 NaN       /m/022g44  
450667                  /m/0g8ngmj                 NaN      /m/0g8ngmm  
450668                  /m/0g8ngmw                 NaN      /m/0btz19d  

[450669 rows x 13 columns]

In [7]:
# Count the total number of rows
total_rows = character_metadata.shape[0]

# Count the number of unique ActorName entries
unique_actor_names = character_metadata['ActorName'].nunique()

# Check for uniqueness
if total_rows == unique_actor_names:
    print("ActorName values are unique.")
else:
    print(f"ActorName values are not unique. There are {total_rows - unique_actor_names} duplicate names.")

ActorName values are not unique. There are 316591 duplicate names.


### plot summaries

In [8]:
plot_columns = ['WikipediaMovieID', 'PlotSummary']

plot_summaries = pd.read_csv(DATA_PATH+'plot_summaries.txt', sep='\t', names=plot_columns)

plot_summaries


WikipediaMovieID                                        PlotSummary
0              23890098  Shlykov, a hard-working taxi driver and Lyosha...
1              31186339  The nation of Panem consists of a wealthy Capi...
2              20663735  Poovalli Induchoodan  is sentenced for six yea...
3               2231378  The Lemon Drop Kid , a New York City swindler,...
4                595909  Seventh-day Adventist Church pastor Michael Ch...
...                 ...                                                ...
42298          34808485  The story is about Reema , a young Muslim scho...
42299           1096473  In 1928 Hollywood, director Leo Andreyev  look...
42300          35102018  American Luthier focuses on Randy Parsons’ tra...
42301           8628195  Abdur Rehman Khan , a middle-aged dry fruit se...
42302           6040782  1940 - Operation Dynamo has just taken place. ...

[42303 rows x 2 columns]

### name clusters

In [9]:
name_clusters_columns = ['FreebaseCharacterActorMapID', 'CharacterName']

name_clusters = pd.read_csv(DATA_PATH+'name.clusters.txt', sep='\t', names=name_clusters_columns)

name_clusters


FreebaseCharacterActorMapID CharacterName
0                  Stuart Little     /m/0k3w9c
1                  Stuart Little     /m/0k3wcx
2                  Stuart Little     /m/0k3wbn
3                       John Doe     /m/0jyg35
4                       John Doe     /m/0k2_zn
...                          ...           ...
2661                  John Rolfe     /m/0k5_ql
2662                  John Rolfe    /m/02vd6vs
2663             Elizabeth Swann     /m/0k1xvz
2664             Elizabeth Swann     /m/0k1x_d
2665             Elizabeth Swann     /m/0k1xxy

[2666 rows x 2 columns]

### TV tropes clusters

In [10]:
tvtropes_columns = ['FreebaseCharacterActorMapID', 'CharacterType']

tvtropes_clusters = pd.read_csv(DATA_PATH+'tvtropes.clusters.txt', sep='\t', names=tvtropes_columns)

tvtropes_clusters


FreebaseCharacterActorMapID  \
0       absent_minded_professor   
1       absent_minded_professor   
2       absent_minded_professor   
3       absent_minded_professor   
4       absent_minded_professor   
..                          ...   
496                   young_gun   
497                   young_gun   
498                   young_gun   
499                   young_gun   
500                   young_gun   

                                         CharacterType  
0    {"char": "Professor Philip Brainard", "movie":...  
1    {"char": "Professor Keenbean", "movie": "Richi...  
2    {"char": "Dr. Reinhardt Lane", "movie": "The S...  
3    {"char": "Dr. Harold Medford", "movie": "Them!...  
4    {"char": "Daniel Jackson", "movie": "Stargate"...  
..                                                 ...  
496  {"char": "Morgan Earp", "movie": "Tombstone", ...  
497  {"char": "Colorado Ryan", "movie": "Rio Bravo"...  
498  {"char": "Tom Sawyer", "movie": "The League of...  
499  {"char": "William H. 'Billy the Kid' Bonney", ...  
500  {"char": "Jake", "movie": "Silverado", "id": "...  

[501 rows x 2 columns]

## Zaynab's part

---
## Lylia's part

---
## Ali's part

### Verifying the matching between **movie_metadata** and **plot_summaries**

In [11]:
# Verify if all WikipediaMovieID in plot_summaries are also in movie_metadata
all_ids_in_metadata = plot_summaries['WikipediaMovieID'].isin(movie_metadata['WikipediaMovieID']).all()

print(f"All WikipediaMovieID in plot_summaries are in movie_metadata: {all_ids_in_metadata}")

All WikipediaMovieID in plot_summaries are in movie_metadata: False


In [12]:
# Calculate the percentage of common WikipediaMovieID between plot_summaries and movie_metadata
common_ids_count = plot_summaries['WikipediaMovieID'].isin(movie_metadata['WikipediaMovieID']).sum()
percentage_common_ids = (common_ids_count / len(plot_summaries)) * 100

print(f"Percentage of common WikipediaMovieID between plot_summaries and movie_metadata: {percentage_common_ids:.2f}%")
print(common_ids_count)
print(len(plot_summaries))
print(len(movie_metadata))

Percentage of common WikipediaMovieID between plot_summaries and movie_metadata: 99.77%
42204
42303
81741


--> Almost all the movies in plot_summaries are in movie_metadata

### Find the bounds of the movies dates

#### With movie_metadata

In [13]:
# Create a copy of the original DataFrame
movie_metadata_copy0 = movie_metadata.copy()

# Drop any rows where ReleaseDate could not be converted
movie_metadata_copy0 = movie_metadata_copy0.dropna(subset=['ReleaseDate'])

# Find minimum and maximum dates after ensuring all dates are converted correctly
min_date = movie_metadata_copy0['ReleaseDate'].min()
max_date = movie_metadata_copy0['ReleaseDate'].max()

print(f"Lower bound (earliest release date): {min_date}")
print(f"Upper bound (latest release date): {max_date}")

Lower bound (earliest release date): 1010.0
Upper bound (latest release date): 2016.0


#### With character_metadata

In [14]:
# Create a copy of the original DataFrame
character_metadata0 = character_metadata.copy()


# Drop rows with NaT in 'ReleaseDate' to avoid issues with min/max
character_metadata0 = character_metadata0.dropna(subset=['ReleaseDate'])

# Get the minimum and maximum release dates
min_date = character_metadata0['ReleaseDate'].min()
max_date = character_metadata0['ReleaseDate'].max()

print(f"Lower bound (earliest release date): {min_date}")
print(f"Upper bound (latest release date): {max_date}")

Lower bound (earliest release date): 1010.0
Upper bound (latest release date): 2016.0


## PART A : Cleaning of CelebA and Rotten Tomatoes datasets

### **Cleaning CelebA dataset**

In [15]:
# Load the file, skipping the first two lines as they are headers or metadata
file_path = './data/MovieSummaries/list_identity_celeba.txt'
identity_data = pd.read_csv(file_path, sep='\s+', skiprows=2, names=['ImageID', 'IdentityName'])
# Replace underscores with spaces in the IdentityName column
identity_data['IdentityName'] = identity_data['IdentityName'].str.replace('_', ' ')
identity_data.head(10)

ImageID         IdentityName
0  000001.jpg  Elizabeth Gutierrez
1  000002.jpg           Emilia Fox
2  000003.jpg         Shane Harper
3  000004.jpg        Leonor Varela
4  000005.jpg     Tatana Kucharova
5  000006.jpg        Jaimee Grubbs
6  000007.jpg     Stephen Colletti
7  000008.jpg        Mario Cantone
8  000009.jpg      Gabriela Spanic
9  000010.jpg          Anita Briem

In [16]:
attr_file_path = './data/MovieSummaries/list_attr_celeba.txt'
# Read the first line as column names and the remaining lines as data
attributes_data = pd.read_csv(attr_file_path, sep='\s+', skiprows=1)

attributes_data.head(10)

ImageID  5_o_Clock_Shadow  Arched_Eyebrows  Attractive  Bags_Under_Eyes  \
0  000001.jpg                -1                1           1               -1   
1  000002.jpg                -1               -1          -1                1   
2  000003.jpg                -1               -1          -1               -1   
3  000004.jpg                -1               -1           1               -1   
4  000005.jpg                -1                1           1               -1   
5  000006.jpg                -1                1           1               -1   
6  000007.jpg                 1               -1           1                1   
7  000008.jpg                 1                1          -1                1   
8  000009.jpg                -1                1           1               -1   
9  000010.jpg                -1               -1           1               -1   

   Bald  Bangs  Big_Lips  Big_Nose  Black_Hair  ...  Sideburns  Smiling  \
0    -1     -1        -1        -1          -1  ...         -1        1   
1    -1     -1        -1         1          -1  ...         -1        1   
2    -1     -1         1        -1          -1  ...         -1       -1   
3    -1     -1        -1        -1          -1  ...         -1       -1   
4    -1     -1         1        -1          -1  ...         -1       -1   
5    -1     -1         1        -1          -1  ...         -1       -1   
6    -1     -1         1         1           1  ...         -1       -1   
7    -1     -1         1        -1           1  ...         -1       -1   
8    -1      1         1        -1          -1  ...         -1        1   
9    -1     -1        -1        -1          -1  ...         -1       -1   

   Straight_Hair  Wavy_Hair  Wearing_Earrings  Wearing_Hat  Wearing_Lipstick  \
0              1         -1                 1           -1                 1   
1             -1         -1                -1           -1                -1   
2             -1          1                -1           -1                -1   
3              1         -1                 1           -1                 1   
4             -1         -1                -1           -1                 1   
5             -1          1                 1           -1                 1   
6              1         -1                -1           -1                -1   
7             -1         -1                -1           -1                -1   
8             -1         -1                 1           -1                 1   
9             -1          1                -1           -1                 1   

   Wearing_Necklace  Wearing_Necktie  Young  
0                -1               -1      1  
1                -1               -1      1  
2                -1               -1      1  
3                 1               -1      1  
4                -1               -1      1  
5                -1               -1      1  
6                -1               -1      1  
7                -1               -1      1  
8                -1               -1      1  
9                -1               -1      1  

[10 rows x 41 columns]

In [17]:
# Merge the two dataframes on 'image_id'
merged_df = pd.merge(identity_data, attributes_data, on='ImageID')

# Display the merged dataframe
merged_df.head()

ImageID         IdentityName  5_o_Clock_Shadow  Arched_Eyebrows  \
0  000001.jpg  Elizabeth Gutierrez                -1                1   
1  000002.jpg           Emilia Fox                -1               -1   
2  000003.jpg         Shane Harper                -1               -1   
3  000004.jpg        Leonor Varela                -1               -1   
4  000005.jpg     Tatana Kucharova                -1                1   

   Attractive  Bags_Under_Eyes  Bald  Bangs  Big_Lips  Big_Nose  ...  \
0           1               -1    -1     -1        -1        -1  ...   
1          -1                1    -1     -1        -1         1  ...   
2          -1               -1    -1     -1         1        -1  ...   
3           1               -1    -1     -1        -1        -1  ...   
4           1               -1    -1     -1         1        -1  ...   

   Sideburns  Smiling  Straight_Hair  Wavy_Hair  Wearing_Earrings  \
0         -1        1              1         -1                 1   
1         -1        1             -1         -1                -1   
2         -1       -1             -1          1                -1   
3         -1       -1              1         -1                 1   
4         -1       -1             -1         -1                -1   

   Wearing_Hat  Wearing_Lipstick  Wearing_Necklace  Wearing_Necktie  Young  
0           -1                 1                -1               -1      1  
1           -1                -1                -1               -1      1  
2           -1                -1                -1               -1      1  
3           -1                 1                 1               -1      1  
4           -1                 1                -1               -1      1  

[5 rows x 42 columns]

Now we have 202,599 images of 10,177 unique actors, so there's many actors with multiple images. In fact, in the scope of our study, we want only 1 attributes description per actor, and hence only a subset of the given attributes. We only care for their physical long-lasting attributes, that remain fix when goig from a picture to another for the same actor. We'll do te following (we could've doen it in reverse order also):

--> we san drop the ImageID as it won't serve us anymore

--> remove duplicates, keep only 1 picture per actor, reducing our dataframe from 202,599 rows to 10,177 rows

--> keep only essential and non-superficial features, that are really proper to each actor (with some marge of error of course)

In [18]:
# Step 1: Drop the 'image_id' column
merged_df2 = merged_df.drop(columns=['ImageID'])

# Step 2: Remove duplicates, keeping only one entry per actor
# We use 'identity_name' as the unique identifier and keep the first occurrence
unique_actor_df = merged_df2.drop_duplicates(subset='IdentityName', keep='first')

# Step 3: Keep only essential, non-superficial features
# Define the list of long-lasting physical attributes we want to keep
essential_attributes = [
    'IdentityName', 'Arched_Eyebrows', 'Attractive', 'Bags_Under_Eyes', 'Bald', 'Bangs', 
    'Big_Lips', 'Big_Nose', 'Black_Hair', 'Blond_Hair', 'Brown_Hair', 
    'Bushy_Eyebrows', 'Chubby', 'Double_Chin', 'Goatee', 'Gray_Hair', 
    'High_Cheekbones', 'Male', 'Mustache', 
    'Narrow_Eyes', 'No_Beard', 'Oval_Face', 'Pale_Skin', 'Pointy_Nose', 'Receding_Hairline', 
    'Rosy_Cheeks', 'Sideburns', 'Straight_Hair', 'Wavy_Hair'
]

# Select only the essential columns from the dataframe
final_df = unique_actor_df[essential_attributes]

# Display the final dataframe
final_df.head(10)

IdentityName  Arched_Eyebrows  Attractive  Bags_Under_Eyes  Bald  \
0  Elizabeth Gutierrez                1           1               -1    -1   
1           Emilia Fox               -1          -1                1    -1   
2         Shane Harper               -1          -1               -1    -1   
3        Leonor Varela               -1           1               -1    -1   
4     Tatana Kucharova                1           1               -1    -1   
5        Jaimee Grubbs                1           1               -1    -1   
6     Stephen Colletti               -1           1                1    -1   
7        Mario Cantone                1          -1                1    -1   
8      Gabriela Spanic                1           1               -1    -1   
9          Anita Briem               -1           1               -1    -1   

   Bangs  Big_Lips  Big_Nose  Black_Hair  Blond_Hair  ...  Narrow_Eyes  \
0     -1        -1        -1          -1          -1  ...           -1   
1     -1        -1         1          -1          -1  ...           -1   
2     -1         1        -1          -1          -1  ...            1   
3     -1        -1        -1          -1          -1  ...           -1   
4     -1         1        -1          -1          -1  ...            1   
5     -1         1        -1          -1          -1  ...           -1   
6     -1         1         1           1          -1  ...           -1   
7     -1         1        -1           1          -1  ...           -1   
8      1         1        -1          -1          -1  ...           -1   
9     -1        -1        -1          -1          -1  ...           -1   

   No_Beard  Oval_Face  Pale_Skin  Pointy_Nose  Receding_Hairline  \
0         1         -1         -1            1                 -1   
1         1         -1         -1           -1                 -1   
2         1         -1         -1            1                 -1   
3         1         -1         -1            1                 -1   
4         1         -1         -1            1                 -1   
5         1         -1         -1           -1                 -1   
6         1         -1         -1            1                 -1   
7         1         -1         -1            1                 -1   
8         1          1         -1            1                 -1   
9         1         -1         -1           -1                 -1   

   Rosy_Cheeks  Sideburns  Straight_Hair  Wavy_Hair  
0           -1         -1              1         -1  
1           -1         -1             -1         -1  
2           -1         -1             -1          1  
3           -1         -1              1         -1  
4           -1         -1             -1         -1  
5           -1         -1             -1          1  
6           -1         -1              1         -1  
7           -1         -1             -1         -1  
8            1         -1             -1         -1  
9           -1         -1             -1          1  

[10 rows x 29 columns]

### Number of actors we have in **character_metadata**

In [19]:
# Assuming 'character_metadata' contains a column named 'ActorName' with each actor's name
num_actors = character_metadata['ActorName'].nunique()
print(f"Number of unique actors: {num_actors}")

Number of unique actors: 134078


### Number of actors we have in **list_identity_celeba**

In [20]:
# Get all unique identity names
unique_names = identity_data['IdentityName'].unique()
# Display the result
print(f"Number of unique actors: {len(unique_names)}")

Number of unique actors: 10177


### Number of actors in common in **character_metadata** & **list_identity_celeba**

In [21]:
# Common actors to the two dataframes

# Convert both columns to sets for efficient intersection
identity_names_set = set(identity_data['IdentityName'])
actor_names_set = set(character_metadata['ActorName'])

# Find the intersection
common_names = identity_names_set.intersection(actor_names_set)

# Output the number of common names and the names themselves
print(f"Number of names in common: {len(common_names)}")
print(common_names)

Number of names in common: 4597
{'Max Ryan', 'Marianna Hill', 'Allison Mack', 'Alex Wilson', 'Seth MacFarlane', 'Ella Rae Peck', 'Nathalie Kelley', 'Nagesh', 'Kayla Ewell', 'Craig Olejnik', 'Martine Beswick', 'Robin Gibb', 'Yanni', 'Elaine Joyce', 'Conrad Bain', 'Marina Berti', 'Ian Charleson', 'Eddie Murphy', 'Hyun Bin', 'Christine Lahti', 'Ray Milland', 'Jessica Steen', 'Enrique Murciano', 'Shalom Harlow', 'Miho Nakayama', 'Susan Flannery', 'Tony Danza', 'Rosalba Neri', 'Brandon Novak', 'Eva Bartok', 'Robby Krieger', 'Andie MacDowell', 'Mimi Rogers', 'James Marsters', 'Yang Mi', 'Travis Pastrana', 'Steven Wilson', 'Angela Little', 'Troy Beyer', 'Sean Lennon', 'Erik von Detten', 'Lee Marvin', 'Jean Simmons', 'Lindy Booth', 'Martha Higareda', 'Sendhil Ramamurthy', 'Julia Stiles', 'Charmion King', 'Judith Light', 'Marta Zmuda', 'Josh Holloway', 'Rebecca Romijn', 'Kathryn Joosten', 'Nikki Harrup', 'Dana Reeve', 'Joanne Kelly', 'Persia White', 'Billie Piper', 'Jenny Levine', 'Yvonne De Ca

### Final CelebA dataframe cleaned to use : **final_celebA_df**

In [22]:
# Convert common_names to a set if it's not already (assuming it’s a set as given)
# Filter final_df to only include rows where 'identity_name' is in common_names
final_celebA_df = final_df[final_df['IdentityName'].isin(common_names)].reset_index(drop=True)

# Display the resulting dataframe
print(f"Number of rows in final_celebA_df: {len(final_celebA_df)}")
final_celebA_df.head(5)

Number of rows in final_celebA_df: 4597


IdentityName  Arched_Eyebrows  Attractive  Bags_Under_Eyes  Bald  \
0  Elizabeth Gutierrez                1           1               -1    -1   
1           Emilia Fox               -1          -1                1    -1   
2         Shane Harper               -1          -1               -1    -1   
3        Leonor Varela               -1           1               -1    -1   
4     Stephen Colletti               -1           1                1    -1   

   Bangs  Big_Lips  Big_Nose  Black_Hair  Blond_Hair  ...  Narrow_Eyes  \
0     -1        -1        -1          -1          -1  ...           -1   
1     -1        -1         1          -1          -1  ...           -1   
2     -1         1        -1          -1          -1  ...            1   
3     -1        -1        -1          -1          -1  ...           -1   
4     -1         1         1           1          -1  ...           -1   

   No_Beard  Oval_Face  Pale_Skin  Pointy_Nose  Receding_Hairline  \
0         1         -1         -1            1                 -1   
1         1         -1         -1           -1                 -1   
2         1         -1         -1            1                 -1   
3         1         -1         -1            1                 -1   
4         1         -1         -1            1                 -1   

   Rosy_Cheeks  Sideburns  Straight_Hair  Wavy_Hair  
0           -1         -1              1         -1  
1           -1         -1             -1         -1  
2           -1         -1             -1          1  
3           -1         -1              1         -1  
4           -1         -1              1         -1  

[5 rows x 29 columns]

### **Cleaning Rotten Tomatoes dataset**

--> Actually we only need "rotten_tomatoes_movies.csv", the critics are non-necessary

In [23]:
# 1. Load the Rotten Tomatoes movies dataset
rt_movies_path = './data/MovieSummaries/rotten_tomatoes_movies.csv'
rt_movies_df = pd.read_csv(rt_movies_path)

# 2. First filter on the columns
rt_movies_df = rt_movies_df[['movie_title', 'critics_consensus', 'content_rating', 
                             'genres', 'actors', 'original_release_date', 
                             'tomatometer_status', 'tomatometer_rating', 
                             'audience_status', 'audience_rating']]

print(rt_movies_df.shape)
rt_movies_df.head(5)

(17712, 10)


movie_title  \
0  Percy Jackson & the Olympians: The Lightning T...   
1                                        Please Give   
2                                                 10   
3                    12 Angry Men (Twelve Angry Men)   
4                       20,000 Leagues Under The Sea   

                                   critics_consensus content_rating  \
0  Though it may seem like just another Harry Pot...             PG   
1  Nicole Holofcener's newest might seem slight i...              R   
2  Blake Edwards' bawdy comedy may not score a pe...              R   
3  Sidney Lumet's feature debut is a superbly wri...             NR   
4  One of Disney's finest live-action adventures,...              G   

                                              genres  \
0  Action & Adventure, Comedy, Drama, Science Fic...   
1                                             Comedy   
2                                    Comedy, Romance   
3                                    Classics, Drama   
4           Action & Adventure, Drama, Kids & Family   

                                              actors original_release_date  \
0  Logan Lerman, Brandon T. Jackson, Alexandra Da...            2010-02-12   
1  Catherine Keener, Amanda Peet, Oliver Platt, R...            2010-04-30   
2  Dudley Moore, Bo Derek, Julie Andrews, Robert ...            1979-10-05   
3  Martin Balsam, John Fiedler, Lee J. Cobb, E.G....            1957-04-13   
4  James Mason, Kirk Douglas, Paul Lukas, Peter L...            1954-01-01   

  tomatometer_status  tomatometer_rating audience_status  audience_rating  
0             Rotten                49.0         Spilled             53.0  
1    Certified-Fresh                87.0         Upright             64.0  
2              Fresh                67.0         Spilled             53.0  
3    Certified-Fresh               100.0         Upright             97.0  
4              Fresh                89.0         Upright             74.0

In [24]:
type(rt_movies_df['original_release_date'][0])

str

In [25]:
# 3. Filter for movies released between 1888 and 2016 (inclusive)
# Convert original_release_date to datetime format for filtering
rt_movies_df['original_release_date'] = rt_movies_df['original_release_date'].astype(str).str.extract(r'(\d{4})')
rt_movies_df['original_release_date'] = rt_movies_df['original_release_date'].astype(float)

rt_movies_df = rt_movies_df[(rt_movies_df['original_release_date'] >= 1888) & 
                            (rt_movies_df['original_release_date'] <= 2016)]

rt_movies_df.shape

(14522, 10)

In [26]:
# 4. Calculate the percentage of common movies with movie_metadata
common_with_metadata = rt_movies_df[rt_movies_df['movie_title'].isin(movie_metadata['MovieName'])]
percentage_common_metadata0 = (len(common_with_metadata) / len(movie_metadata)) * 100
percentage_common_metadata1 = (len(common_with_metadata) / len(rt_movies_df)) * 100
print(f"Percentage of common movies with movie_metadata (wrt to movie_metadata): {percentage_common_metadata0:.2f}%")
print(f"Percentage of common movies with movie_metadata (wrt to rt_movies_df): {percentage_common_metadata1:.2f}%")

Percentage of common movies with movie_metadata (wrt to movie_metadata): 11.99%
Percentage of common movies with movie_metadata (wrt to rt_movies_df): 67.48%


### Final Rotten Tomatoes dataframe cleaned to use : **final_RottenTomatoes_df**

--> Keeping only the MOVIE TITLE, ACTORS, TOMATOMETER RATING and AUDIENCE RATING. With only the common movies with **movie_metadata**.

In [27]:
# Second filter on the columns
RottenTomatoes_df_filtered = rt_movies_df[['movie_title', 'actors', 'tomatometer_rating', 'audience_rating']]

# Filter to include only common movies between RottenTomatoes_df and movie_metadata
common_movies = RottenTomatoes_df_filtered[
    RottenTomatoes_df_filtered['movie_title'].isin(movie_metadata['MovieName'])
]

# Create final DataFrame
final_RottenTomatoes_df = common_movies.reset_index(drop=True)

# Display the result
final_RottenTomatoes_df.head(10)

movie_title  \
0  Percy Jackson & the Olympians: The Lightning T...   
1                                        Please Give   
2                                                 10   
3                                       The 39 Steps   
4                                       3:10 to Yuma   
5                                    Abraham Lincoln   
6                                         Dark Water   
7                                        The Accused   
8                                      The Lost City   
9                                 The Breaking Point   

                                              actors  tomatometer_rating  \
0  Logan Lerman, Brandon T. Jackson, Alexandra Da...                49.0   
1  Catherine Keener, Amanda Peet, Oliver Platt, R...                87.0   
2  Dudley Moore, Bo Derek, Julie Andrews, Robert ...                67.0   
3  Robert Donat, Madeleine Carroll, Godfrey Tearl...                96.0   
4  Glenn Ford, Van Heflin, Felicia Farr, Leora Da...                96.0   
5  Walter Huston, Una Merkel, Kay Hammond, Ian Ke...                82.0   
6  Hitomi Kuroki, Rio Kanno, Shigemitsu Ogi, Mire...                80.0   
7  Jodie Foster, Kelly McGillis, Bernie Coulson, ...                91.0   
8  Andy Garcia, Dustin Hoffman, Bill Murray, Inés...                25.0   
9  John Garfield, Patricia Neal, Phyllis Thaxter,...               100.0   

   audience_rating  
0             53.0  
1             64.0  
2             53.0  
3             86.0  
4             79.0  
5             40.0  
6             66.0  
7             79.0  
8             64.0  
9             86.0

## **1. Defining "Success" metrics**

--> We want to first establish clear and quatifiable measures of an actor's success :

- Box office revenue : **total & average revenue of movies** an actor has participated in
- Populatity : **number of films**
- Critical Acclaim : 

Dataframes : plot_summaries, movie_metadata, character_metadata, tvtropes_clusters

### Total revenue of movies an actor has participated in

In [28]:
# Verify if in movie_metadata['BoxOfficeRevenue'] there's some nan values, to maybe clean
nan_percentage = movie_metadata['BoxOfficeRevenue'].isna().mean() * 100
print(f"Percentage of NaNs in 'BoxOfficeRevenue': {nan_percentage:.2f}%")

# Replace NaN values in BoxOfficeRevenue with 0
# movie_metadata['BoxOfficeRevenue'] = movie_metadata['BoxOfficeRevenue'].fillna(0)

Percentage of NaNs in 'BoxOfficeRevenue': 89.72%


With 89.72% of values missing in the BoxOfficeRevenue column, filling with zero remains a viable option. However, the high proportion of missing data suggests that relying on box office revenue alone might not provide a complete or representative measure of success. **We'll try later to complete the given data on BoxOfficeRevenue bringing external data from Wikipedia.**

--> CONTINUITY AFTER IMPORTING A MAX TO FILL THE MISSING DATA

In [29]:
# Merge movie and character data on WikipediaMovieID
merged_df = pd.merge(character_metadata, movie_metadata, on='WikipediaMovieID', how='inner')
merged_df

WikipediaMovieID FreebaseMovieID_x  ReleaseDate_x  \
0                 975900         /m/03vyhn         2001.0   
1                 975900         /m/03vyhn         2001.0   
2                 975900         /m/03vyhn         2001.0   
3                 975900         /m/03vyhn         2001.0   
4                 975900         /m/03vyhn         2001.0   
...                  ...               ...            ...   
450664            913762         /m/03pcrp         1992.0   
450665            913762         /m/03pcrp         1992.0   
450666          28308153        /m/0cp05t9         1957.0   
450667          28308153        /m/0cp05t9         1957.0   
450668          28308153        /m/0cp05t9         1957.0   

                     CharacterName    ActorDOB ActorGender  ActorHeight  \
0                         Akooshay  1958-08-26           F        1.620   
1       Lieutenant Melanie Ballard  1974-08-15           F        1.780   
2              Desolation Williams  1969-06-15           M        1.727   
3               Sgt Jericho Butler  1967-09-12           M        1.750   
4                  Bashira Kincaid  1977-09-25           F        1.650   
...                            ...         ...         ...          ...   
450664                      Elensh     1970-05           F          NaN   
450665                      Hibiki  1965-04-12           M          NaN   
450666                         NaN  1941-11-18           M        1.730   
450667                         NaN         NaN         NaN          NaN   
450668                         NaN         NaN         NaN          NaN   

       ActorEthnicity           ActorName  ActorAgeAtRelease  ...  \
0                 NaN      Wanda De Jesus               42.0  ...   
1          /m/044038p  Natasha Henstridge               27.0  ...   
2             /m/0x67            Ice Cube               32.0  ...   
3                 NaN       Jason Statham               33.0  ...   
4                 NaN         Clea DuVall               23.0  ...   
...               ...                 ...                ...  ...   
450664            NaN  Dorothy Elias-Fahn                NaN  ...   
450665            NaN       Jonathan Fahn               27.0  ...   
450666      /m/02w7gg      David Hemmings               15.0  ...   
450667            NaN    Roberta Paterson                NaN  ...   
450668            NaN         John Rogers                NaN  ...   

       FreebaseCharacterID FreebaseActorID FreebaseMovieID_y  \
0               /m/0bgcj3x      /m/03wcfv7         /m/03vyhn   
1               /m/0bgchn4       /m/0346l4         /m/03vyhn   
2               /m/0bgchn_      /m/01vw26l         /m/03vyhn   
3               /m/0bgchnq       /m/034hyc         /m/03vyhn   
4               /m/0bgchp9       /m/01y9xg         /m/03vyhn   
...                    ...             ...               ...   
450664          /m/0kr406h       /m/0b_vcv         /m/03pcrp   
450665          /m/0kr4090       /m/0bx7_j         /m/03pcrp   
450666                 NaN       /m/022g44        /m/0cp05t9   
450667                 NaN      /m/0g8ngmm        /m/0cp05t9   
450668                 NaN      /m/0btz19d        /m/0cp05t9   

                                                MovieName ReleaseDate_y  \
0                                          Ghosts of Mars        2001.0   
1                                          Ghosts of Mars        2001.0   
2                                          Ghosts of Mars        2001.0   
3                                          Ghosts of Mars        2001.0   
4                                          Ghosts of Mars        2001.0   
...                                                   ...           ...   
450664  The Super Dimension Fortress Macross II: Lover...        1992.0   
450665  The Super Dimension Fortress Macross II: Lover...        1992.0   
450666                              Five Clues to Fortune        1957.0   
450667                    

In [30]:
# Calculate Total Box Office Revenue per Actor
actor_revenue = merged_df.groupby('ActorName')['BoxOfficeRevenue'].sum().reset_index()
total_revenue = actor_revenue.rename(columns={'BoxOfficeRevenue': 'TotalBoxOfficeRevenue'})
# Sort total_revenue DataFrame in place by 'TotalBoxOfficeRevenue' in descending order
total_revenue.sort_values(by='TotalBoxOfficeRevenue', ascending=False, inplace=True)
print (total_revenue.head(20))

                   ActorName  TotalBoxOfficeRevenue
129501         Warwick Davis           1.293016e+10
111587     Samuel L. Jackson           1.278943e+10
39968           Frank Welker           1.028744e+10
2220            Alan Rickman           1.020871e+10
106644       Robbie Coltrane           1.009465e+10
24311          Conrad Vernon           9.786402e+09
123933             Tom Hanks           9.623361e+09
41629            Gary Oldman           9.614965e+09
79198           Maggie Smith           9.280352e+09
49940           Hugo Weaving           8.896132e+09
33130           Eddie Murphy           8.682898e+09
61841      John Ratzenberger           8.414363e+09
123893            Tom Felton           8.332480e+09
106915     Robert Downey Jr.           8.056541e+09
43035   Geraldine Somerville           7.865889e+09
35612            Emma Watson           7.865083e+09
26490       Daniel Radcliffe           7.863063e+09
123845            Tom Cruise           7.861340e+09
48065   Hele

In [31]:
# Count actors with 0.0 total revenue
zero_revenue_count = actor_revenue[actor_revenue['BoxOfficeRevenue'] == 0.0].shape[0]

# Calculate total number of actors
total_actors = actor_revenue.shape[0]

# Calculate percentage
percentage_zero_revenue = (zero_revenue_count / total_actors) * 100
print(f"Percentage of actors with a total revenue of 0.0: {percentage_zero_revenue:.2f}%")

Percentage of actors with a total revenue of 0.0: 72.12%


Ok so that's maybe not so good to base ourself on this metric, with approx. 72.12% of lost values. Or maybe not, analysis may be biased towards actors in high-grossing films, ignoring those in smaller projects.

### Average revenue of movies an actor has participated in

In [32]:
# Calculate Average Box Office Revenue per Movie (per actor)
average_revenue = merged_df.groupby('ActorName')['BoxOfficeRevenue'].mean().reset_index().rename(columns={'BoxOfficeRevenue': 'AverageBoxOfficeRevenue'})
average_revenue.sort_values(by='AverageBoxOfficeRevenue', ascending=False, inplace=True)
average_revenue.head(20)

ActorName  AverageBoxOfficeRevenue
55768                 Jason Whyte             2.782275e+09
112919             Scott Lawrence             2.782275e+09
113091         Sean Anthony Moran             2.782275e+09
74868             Lewis Abernathy             2.185372e+09
30556                  Dileep Rao             1.803904e+09
70482            Kirill Nikiforov             1.511758e+09
3595               Alexis Denisof             1.511758e+09
78604           M'laah Kaur Singh             1.511758e+09
23846              Cobie Smulders             1.511758e+09
103669             Rashmi Rustagi             1.511758e+09
130022                  Will Dunn             1.328111e+09
34876          Ellie Darcey-Alden             1.328111e+09
110283                Ryan Turner             1.328111e+09
12389             Benedict Clarke             1.328111e+09
3643               Alfie McIlwain             1.328111e+09
27022         Daphne de Beistegui             1.328111e+09
9572                 Arthur Bowen             1.328111e+09
47696              Hebe Beardsall             1.328111e+09
48062               Helena Barlow             1.328111e+09
109189  Rosie Huntington-Whiteley             1.123747e+09

In [33]:
# Calculate Number of Movies per Actor
movie_count = merged_df.groupby('ActorName')['MovieName'].nunique().reset_index().rename(columns={'MovieName': 'NumberOfMovies'})
# Sort movie_count DataFrame in place by 'NumberOfMovies' in descending order
movie_count.sort_values(by='NumberOfMovies', ascending=False, inplace=True)
movie_count.head(20)

ActorName  NumberOfMovies
86136            Mel Blanc             573
90201   Mithun Chakraborty             322
95972         Oliver Hardy             299
90474             Mohanlal             234
90355           Moe Howard             225
79708            Mammootty             225
72567           Larry Fine             219
30021      Dharmendra Deol             215
101832         Prakash Raj             205
15231         Brahmanandam             198
46976         Harold Lloyd             191
53468    Jagathi Sreekumar             191
8805           Anupam Kher             190
5054      Amitabh Bachchan             189
114086       Shakti Kapoor             185
66325         Kamal Haasan             175
62082           John Wayne             173
103052          Rajnikanth             172
92343               Nassar             171
93035        Nedumudi Venu             166

In [34]:
# Combine Metrics
success_metrics = total_revenue.merge(average_revenue, on='ActorName').merge(movie_count, on='ActorName')

# Display top 10 actors by Total Box Office Revenue
top_actors = success_metrics.sort_values(by='TotalBoxOfficeRevenue', ascending=False)
top_actors.head(20)

ActorName  TotalBoxOfficeRevenue  AverageBoxOfficeRevenue  \
0          Warwick Davis           1.293016e+10             6.805349e+08   
1      Samuel L. Jackson           1.278943e+10             1.776310e+08   
2           Frank Welker           1.028744e+10             1.686466e+08   
3           Alan Rickman           1.020871e+10             3.520243e+08   
4        Robbie Coltrane           1.009465e+10             3.364884e+08   
5          Conrad Vernon           9.786402e+09             5.756707e+08   
6              Tom Hanks           9.623361e+09             2.047524e+08   
7            Gary Oldman           9.614965e+09             2.670824e+08   
8           Maggie Smith           9.280352e+09             3.200121e+08   
9           Hugo Weaving           8.896132e+09             3.706721e+08   
10          Eddie Murphy           8.682898e+09             1.929533e+08   
11     John Ratzenberger           8.414363e+09             4.428612e+08   
12            Tom Felton           8.332480e+09             6.943734e+08   
13     Robert Downey Jr.           8.056541e+09             1.790342e+08   
14  Geraldine Somerville           7.865889e+09             7.865889e+08   
15           Emma Watson           7.865083e+09             7.865083e+08   
16      Daniel Radcliffe           7.863063e+09             7.148239e+08   
17            Tom Cruise           7.861340e+09             2.312159e+08   
18  Helena Bonham Carter           7.857267e+09             3.022026e+08   
19          Bruce Willis           7.829828e+09             1.242830e+08   

    NumberOfMovies  
0               28  
1              104  
2              150  
3               41  
4               59  
5               15  
6               53  
7               53  
8               55  
9               36  
10              42  
11              34  
12              15  
13              66  
14              13  
15              13  
16              15  
17              39  
18              56  
19              78

## **2. Define actors' characteristics that we want to evaluate**

### **Demographics** : Height, Gender, Ethnicity

In [35]:
# Calculate percentage of NaN values for ActorGender and ActorHeight
gender_nan_percentage = character_metadata['ActorGender'].isna().mean() * 100
height_nan_percentage = character_metadata['ActorHeight'].isna().mean() * 100
ethnicity_nan_percentage = character_metadata['ActorEthnicity'].isna().mean() * 100

print(f"Percentage of NaNs in 'ActorGender': {gender_nan_percentage:.2f}%")
print(f"Percentage of NaNs in 'ActorHeight': {height_nan_percentage:.2f}%")
print(f"Percentage of NaNs in 'ActorEthnicity': {ethnicity_nan_percentage:.2f}%")

Percentage of NaNs in 'ActorGender': 10.12%
Percentage of NaNs in 'ActorHeight': 65.65%
Percentage of NaNs in 'ActorEthnicity': 76.47%


--> AGAIN, WE'LL TRY TO FILL THE MISSING VALUES IMPORTING FROM WIKI

---
## Christian's part

---
## Yassin's part